### PR TITLE
CSV Handler

### PR DESCRIPTION
**Pull request by vijay-sethu-s**

### CODE REVIEW

File: superagentx_handlers/csv_data.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            llm_client: LLMClient | None = None,
+            prompt: str | None = None
+    ):
+        self.input = csv_path
+        self.llm_client: LLMClient = llm_client
+        self.prompt = prompt
+
+        if not self.llm_client:
+            llm_config = {'llm_type': 'openai'}
+            self.llm_client: LLMClient = LLMClient(llm_config=llm_config)
```
Comment (SyedHabibM) [inline superagentx_handlers/csv_data.py:26]: You can remove type annotation here

File: superagentx_handlers/csv_data.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        Args:
+            query (str):
+                The search term or query string used to retrieve relevant results. This can be a keyword, a phrase,
+                or any other string that specifies what to search for.
+        """
+
+        try:
+            prompt = self.prompt
+            if not prompt:
+                df = pd.read_csv(self.input)
```
Comment (SyedHabibM) [inline superagentx_handlers/csv_data.py:45]: If possible change this with `sync_to_async`, and test it

File: superagentx_handlers/csv_data.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+                }
+            ]
+            chat_completion_params = ChatCompletionParams(
+                messages=messages,
+                temperature=0
+            )
+            response = await self.llm_client.achat_completion(
+                chat_completion_params=chat_completion_params
+            )
+            result = response.choices[0].message.content.strip()
```
Comment (SyedHabibM) [inline superagentx_handlers/csv_data.py:62]: Make sure, `choices` is not none and not empty

File: superagentx_handlers/csv_data.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+            start = '```python\n'
+            end = '```'
+            trim_res = re.findall(re.escape(start) + "(.+?)" + re.escape(end), result, re.DOTALL)
```
Comment (SyedHabibM) [inline superagentx_handlers/csv_data.py:65]: Make sure, `trim_res` is not none and not empty

File: superagentx_handlers/csv_data.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+            return eval(trim_res[0]).to_json()
+        except Exception as ex:
+            message = f"Error while searching result! {ex}"
+            logger.error(message)
+            raise Exception(message)
+
+    def __dir__(self):
```
Comment (SyedHabibM) [inline superagentx_handlers/csv_data.py:72]: Don't need this, instead use `@tool` for `search` method decorator. Refer existing handlers in the dev

File: superagentx_handlers/csv_data.py [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+    def __init__(
+            self,
+            *,
+            csv_path: str,
+            llm_client: LLMClient | None = None,
+            prompt: str | None = None
+    ):
+        self.input = csv_path
+        self.llm_client: LLMClient = llm_client
```
[the remark this answers is not in the document]
Comment (vijay-sethu-s) [reply superagentx_handlers/csv_data.py]: resolved

File: superagentx_handlers/csv_data.py [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+        self.prompt = prompt
+
+        if not self.llm_client:
+            llm_config = {'llm_type': 'openai'}
+            self.llm_client: LLMClient = LLMClient(llm_config=llm_config)
+
+    async def search(self,
```
[the remark this answers is not in the document]
Comment (vijay-sethu-s) [reply superagentx_handlers/csv_data.py]: resolved

File: superagentx_handlers/csv_data.py (hunk 2 above)
Comment (vijay-sethu-s) [reply superagentx_handlers/csv_data.py]: resolved

File: superagentx_handlers/csv_data.py (hunk 3 above)
Comment (vijay-sethu-s) [reply superagentx_handlers/csv_data.py]: resolved

File: superagentx_handlers/csv_data.py (hunk 1 above)
Comment (vijay-sethu-s) [reply superagentx_handlers/csv_data.py]: resolved

File: superagentx_handlers/csv_data.py (hunk 4 above)
Comment (vijay-sethu-s) [reply superagentx_handlers/csv_data.py]: reolved

File: superagentx_handlers/csv_data.py (hunk 5 above)
Comment (vijay-sethu-s) [reply superagentx_handlers/csv_data.py]: resolved

File: superagentx_handlers/csv_data.py [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+logger = logging.getLogger(__name__)
+
+
+class CsvHandler(BaseHandler):
+
+    def __init__(
+            self,
+            *,
+            csv_path: str,
+            llm_client: LLMClient | None = None,
```
Comment (SyedHabibM) [inline superagentx_handlers/csv_data.py:19]: Make this as required. Plz

**Review by SyedHabibM — CHANGES_REQUESTED** [review]

File: superagentx_handlers/csv_data.py (hunk 8 above)
Comment (vijay-sethu-s) [reply superagentx_handlers/csv_data.py]: resolved

**Review by SyedHabibM — APPROVED** [review]

**Pull request merged**